api/src/test/java/org/openmrs/api/impl/ConditionServiceImplTest.java

```diff
@@ -0,0 +1,190 @@
+	 */
+	@Test
+	public void getConditionByUuid_shouldReturnNullIfNoObjectFoundWithGivenUuid() {
+		Assert.assertNull(conditionService.getConditionByUuid("invalid uuid"));
+	}
+
+	/**
+	 * @see ConditionService#getCondition(Integer) 
+	 */
+	@Test
+	public void getConditionById_shouldFindConditionGivenValidId() {
+		executeDataSet(CONDITION_XML);
+		int conditionId = 1;
+		String uuid="2cc6880e-2c46-15e4-9038-a6c5e4d22fb7";
+		String clinical_status = "INACTIVE";
+		String verification_status = "PROVISIONAL";
+		expectedException = ExpectedException.none();
+		Condition condition = conditionService.getCondition(conditionId);
+		Assert.assertEquals(clinical_status, condition.getClinicalStatus().toString());
+		Assert.assertEquals(verification_status, condition.getVerificationStatus().toString());
+		Assert.assertEquals(uuid, condition.getUuid());
+	}
+
+	/**
+	 * @see ConditionService#getActiveConditions(Patient) 
+	 */
+	@Test
+	public void getActiveConditions_shouldGetActiveConditions() {
+		executeDataSet(CONDITION_XML);
+		int patientId = 2;
+		expectedException = ExpectedException.none();
+		List<Condition> activeConditions = conditionService.getActiveConditions(patientService.getPatient(patientId));
+		Condition firstActiveCondition = activeConditions.get(0);
+		Condition secondActiveCondition = activeConditions.get(1);
+		String firstUUID = "2cc6880e-2c46-11e4-9138-a6c5e4d20fb7";
+		String secondUUID = "2cc6880e-2c46-15e4-9038-a6c5e4d22fb7";
+		Assert.assertFalse(activeConditions.isEmpty());
+		Assert.assertTrue(activeConditions.size() == 2);
+		Assert.assertEquals(firstUUID,firstActiveCondition.getUuid());
+		Assert.assertEquals(secondUUID, secondActiveCondition.getUuid());
+	}
+	
+	
+
+	/**
+	 * @see ConditionService#voidCondition(Condition, String) 
+	 */
+	@Test
+	public void voidCondition_shouldVoidConditionSuccessfully(){
+		Condition condition = new Condition();
+		int patientId = 2;
+		condition.setCondition(new CodedOrFreeText());
+		condition.setAdditionalDetail("voiding a condition");
+		condition.setVerificationStatus(ConditionVerificationStatus.PROVISIONAL);
+		condition.setClinicalStatus(ConditionClinicalStatus.INACTIVE);
+		condition.setUuid(UUID);
+		condition.setPatient(patientService.getPatient(patientId));
+		expectedException = ExpectedException.none();
+		conditionService.voidCondition(condition, "Test Reason");
+		Condition voidedCondition = conditionService.getConditionByUuid(UUID);
+		Assert.assertEquals(ConditionVerificationStatus.PROVISIONAL, voidedCondition.getVerificationStatus());
+		Assert.assertEquals(ConditionClinicalStatus.INACTIVE, voidedCondition.getClinicalStatus());
+		Assert.assertEquals(UUID, voidedCondition.getUuid());
+	}
+
+	/**
+	 * @see ConditionService#unvoidCondition(Condition) 
+	 */
+	@Test
+	public void unvoidCondition_shouldUnvoidConditionSuccessfully(){
+		Condition condition = new Condition();
+		int patientId = 2;
+		condition.setCondition(new CodedOrFreeText());
```

Comment:
Just load a voided condition from your dataset file and then use it for unvoiding in this test.

Reply:
Okay